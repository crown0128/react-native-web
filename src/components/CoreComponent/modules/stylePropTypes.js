import { PropTypes } from 'react'

const numberOrString = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string
])

const { string } = PropTypes

export default {
  alignContent: string,
  alignItems: string,
  alignSelf: string,
  backfaceVisibility: string,
  // background
  backgroundAttachment: string,
  backgroundClip: string,
  backgroundColor: string,
  backgroundImage: string,
  backgroundOrigin: string,
  backgroundPosition: string,
  backgroundRepeat: string,
  backgroundSize: string,
  // border color
  borderColor: numberOrString,
  borderBottomColor: numberOrString,
  borderLeftColor: numberOrString,
  borderRightColor: numberOrString,
  borderTopColor: numberOrString,
  // border-radius
  borderRadius: numberOrString,
  borderTopLeftRadius: numberOrString,
  borderTopRightRadius: numberOrString,
  borderBottomLeftRadius: numberOrString,
  borderBottomRightRadius: numberOrString,
  // border style
  borderStyle: numberOrString,
  borderBottomStyle: numberOrString,
  borderLeftStyle: numberOrString,
  borderRightStyle: numberOrString,
  borderTopStyle: numberOrString,
  // border width
  borderWidth: numberOrString,
  borderBottomWidth: numberOrString,
  borderLeftWidth: numberOrString,
  borderRightWidth: numberOrString,
  borderTopWidth: numberOrString,
  bottom: numberOrString,
  boxSizing: string,
  clear: string,
  color: string,
  cursor: string,
  direction: string,
  display: string,
  flexBasis: string,
  flexDirection: string,
  flexGrow: numberOrString,
  flexShrink: numberOrString,
  flexWrap: string,
  float: string,
  font: string,
  fontFamily: string,
  fontSize: string,
  fontStyle: string,
  fontWeight: string,
  height: numberOrString,
  justifyContent: string,
  left: numberOrString,
  letterSpacing: string,
  lineHeight: numberOrString,
  // margin
  margin: numberOrString,
  marginBottom: numberOrString,
  marginLeft: numberOrString,
  marginRight: numberOrString,
  marginTop: numberOrString,
  // min/max
  maxHeight: numberOrString,
  maxWidth: numberOrString,
  minHeight: numberOrString,
  minWidth: numberOrString,
  opacity: numberOrString,
  order: numberOrString,
  overflow: string,
  overflowX: string,
  overflowY: string,
  // padding
  padding: numberOrString,
  paddingBottom: numberOrString,
  paddingLeft: numberOrString,
  paddingRight: numberOrString,
  paddingTop: numberOrString,
  position: string,
  right: numberOrString,
  textAlign: string,
  textDecoration: string,
  textTransform: string,
  top: numberOrString,
  userSelect: string,
  visibility: string,
  whiteSpace: string,
  width: numberOrString,
  wordWrap: string,
  zIndex: numberOrString
}
