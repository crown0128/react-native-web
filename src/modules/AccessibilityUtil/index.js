import propsToAccessibilityComponent from './propsToAccessibilityComponent';
import propsToAriaRole from './propsToAriaRole';
import propsToTabIndex from './propsToTabIndex';

module.exports = {
  propsToAccessibilityComponent,
  propsToAriaRole,
  propsToTabIndex
};
