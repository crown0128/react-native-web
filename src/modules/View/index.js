import { pickProps } from '../filterObjectProps';
import StylePropTypes from '../StylePropTypes';
import React, { PropTypes } from 'react';
import WebStyleComponent from '../WebStyleComponent';

// https://github.com/facebook/css-layout#default-values
const ViewStyleDefaultProps = {
  alignItems: 'stretch',
  borderWidth: 0,
  borderStyle: 'solid',
  boxSizing: 'border-box',
  display: 'flex',
  flexBasis: 'auto',
  flexDirection: 'column',
  flexShrink: 0,
  listStyle: 'none',
  margin: 0,
  padding: 0,
  position: 'relative'
};

const ViewStylePropTypes = {
  ...StylePropTypes.BackgroundPropTypes,
  ...StylePropTypes.BorderThemePropTypes,
  ...StylePropTypes.LayoutPropTypes,
  boxShadow: PropTypes.string,
  opacity: PropTypes.number,
  transform: PropTypes.string
};

class View extends React.Component {
  static _getPropTypes() {
    return {
      className: PropTypes.string,
      element: PropTypes.oneOfType([
        PropTypes.string, PropTypes.func
      ]),
      pointerEvents: PropTypes.oneOf([
        'auto',
        'box-none',
        'box-only',
        'none'
      ]),
      style: PropTypes.shape(ViewStylePropTypes)
    };
  }

  static _getDefaultProps() {
    return {
      className: '',
      element: 'div',
      style: {}
    };
  }

  render() {
    const { className, element, pointerEvents, style, ...other } = this.props;
    const filteredStyle = pickProps(style, Object.keys(ViewStylePropTypes));
    const pointerEventsStyle = pointerEvents && { pointerEvents };
    const mergedStyle = {
      ...ViewStyleDefaultProps,
      ...filteredStyle,
      ...pointerEventsStyle
    };

    return (
      <WebStyleComponent
        {...other}
        className={`View ${className}`}
        element={element}
        style={mergedStyle}
      />
    );
  }
}

View.propTypes = View._getPropTypes();
View.defaultProps = View._getDefaultProps();

export default View;
export { ViewStylePropTypes };
