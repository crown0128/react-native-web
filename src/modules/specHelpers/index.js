import assert from 'assert'
import React from 'react/addons'

const ReactTestUtils = React.addons.TestUtils

export const assertProps = {
  accessibilityLabel: function (Component, props) {
    // with label
    const accessibilityLabel = 'accessibilityLabel'
    const dom = renderToDOM(<Component {...props} accessibilityLabel={accessibilityLabel} />)
    assert.equal(dom.getAttribute('aria-label'), accessibilityLabel)
  },

  accessibilityLiveRegion: function (Component, props) {
    const accessibilityLiveRegion = 'polite'
    const dom = renderToDOM(<Component {...props} accessibilityLiveRegion={accessibilityLiveRegion} />)
    assert.equal(dom.getAttribute('aria-live'), accessibilityLiveRegion)
  },

  accessibilityRole: function (Component, props) {
    const accessibilityRole = 'main'
    const dom = renderToDOM(<Component {...props} accessibilityRole={accessibilityRole} />)
    assert.equal(dom.getAttribute('role'), accessibilityRole)
  },

  accessible: function (Component, props) {
    // accessible
    let dom = renderToDOM(<Component {...props} accessible />)
    assert.equal(dom.getAttribute('aria-hidden'), null)
    // not accessible
    dom = renderToDOM(<Component {...props} accessible={false} />)
    assert.equal(dom.getAttribute('aria-hidden'), 'true')
  },

  component: function (Component, props) {
    const component = 'main'
    const dom = renderToDOM(<Component {...props} component={component} />)
    const tagName = (dom.tagName).toLowerCase()
    assert.equal(tagName, component)
  },

  style: function (Component, props) {
    let shallow
    // default styles
    shallow = shallowRender(<Component {...props} />)
    assert.deepEqual(
      shallow.props.style,
      Component.defaultProps.style
    )
    // filtering of unsupported styles
    const styleToFilter = { unsupported: 'unsupported' }
    shallow = shallowRender(<Component {...props} style={styleToFilter} />)
    assert.deepEqual(
      shallow.props.style.unsupported,
      undefined,
      'unsupported "style" properties must not be transferred'
    )
    // merging of custom styles
    const styleToMerge = { margin: '10' }
    shallow = shallowRender(<Component {...props} style={styleToMerge} />)
    assert.deepEqual(
      shallow.props.style.margin,
      styleToMerge.margin,
    )
  },

  testID: function (Component, props) {
    // no testID
    let dom = renderToDOM(<Component {...props} />)
    assert.equal(dom.getAttribute('data-testid'), null)
    // with testID
    const testID = 'Example.testID'
    dom = renderToDOM(<Component {...props} testID={testID} />)
    assert.equal(dom.getAttribute('data-testid'), testID)
  }
}

export function render(element, container) {
  return container
    ? React.render(element, container)
    : ReactTestUtils.renderIntoDocument(element)
}

export function renderToDOM(element, container) {
  const result = render(element, container)
  return React.findDOMNode(result)
}

export function shallowRender(component, context = {}) {
  const shallowRenderer = React.addons.TestUtils.createRenderer()
  shallowRenderer.render(component, context)
  return shallowRenderer.getRenderOutput()
}
