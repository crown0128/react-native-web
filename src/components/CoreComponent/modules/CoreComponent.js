import React, { PropTypes } from 'react'
import restyle from './restyle'
import stylePropTypes from './stylePropTypes'

class CoreComponent extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    component: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.string
    ]),
    style: PropTypes.object,
    testID: PropTypes.string
  }

  static stylePropTypes = stylePropTypes;

  static defaultProps = {
    className: '',
    component: 'div'
  }

  render() {
    const {
      className,
      component: Component,
      style,
      testID,
      ...other
    } = this.props

    return (
      <Component
        {...other}
        {...restyle({ className, style })}
        data-testid={process.env.NODE_ENV === 'production' ? null : testID}
      />
    )
  }
}

export default CoreComponent
