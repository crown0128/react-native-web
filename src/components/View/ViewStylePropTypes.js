import { pickProps } from '../../modules/filterObjectProps'
import CoreComponent from '../CoreComponent'

export default {
  ...pickProps(CoreComponent.stylePropTypes, [
    'alignContent',
    'alignItems',
    'alignSelf',
    'backfaceVisibility',
    // background
    'backgroundAttachment',
    'backgroundClip',
    'backgroundColor',
    'backgroundImage',
    'backgroundPosition',
    'backgroundOrigin',
    'backgroundRepeat',
    'backgroundSize',
    // border-color
    'borderColor',
    'borderTopColor',
    'borderRightColor',
    'borderBottomColor',
    'borderLeftColor',
    // border-radius
    'borderRadius',
    'borderTopLeftRadius',
    'borderTopRightRadius',
    'borderBottomLeftRadius',
    'borderBottomRightRadius',
    // border style
    'borderStyle',
    'borderBottomStyle',
    'borderLeftStyle',
    'borderRightStyle',
    'borderTopStyle',
    // border width
    'borderWidth',
    'borderBottomWidth',
    'borderLeftWidth',
    'borderRightWidth',
    'borderTopWidth',
    'bottom',
    'boxShadow',
    'boxSizing',
    'cursor',
    'flexBasis',
    'flexDirection',
    'flexGrow',
    'flexShrink',
    'flexWrap',
    'height',
    'justifyContent',
    'left',
    // margin
    'margin',
    'marginBottom',
    'marginLeft',
    'marginRight',
    'marginTop',
    // max/min
    'maxHeight',
    'maxWidth',
    'minHeight',
    'minWidth',
    'opacity',
    'order',
    'overflow',
    'overflowX',
    'overflowY',
    // padding
    'padding',
    'paddingBottom',
    'paddingLeft',
    'paddingRight',
    'paddingTop',
    'position',
    'right',
    'top',
    'transform',
    'userSelect',
    'visibility',
    'width',
    'zIndex'
  ])
}
