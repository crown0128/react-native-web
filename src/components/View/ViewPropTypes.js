import BaseComponentPropTypes from '../../propTypes/BaseComponentPropTypes';
import EdgeInsetsPropType from '../../propTypes/EdgeInsetsPropType';
import StyleSheetPropType from '../../propTypes/StyleSheetPropType';
import ViewStylePropTypes from './ViewStylePropTypes';
import { PropTypes } from 'react';

const ViewPropTypes = {
  ...BaseComponentPropTypes,
  children: PropTypes.any,
  collapsable: PropTypes.bool,
  hitSlop: EdgeInsetsPropType,
  onClick: PropTypes.func,
  onClickCapture: PropTypes.func,
  onLayout: PropTypes.func,
  onMoveShouldSetResponder: PropTypes.func,
  onMoveShouldSetResponderCapture: PropTypes.func,
  onResponderGrant: PropTypes.func,
  onResponderMove: PropTypes.func,
  onResponderReject: PropTypes.func,
  onResponderRelease: PropTypes.func,
  onResponderTerminate: PropTypes.func,
  onResponderTerminationRequest: PropTypes.func,
  onStartShouldSetResponder: PropTypes.func,
  onStartShouldSetResponderCapture: PropTypes.func,
  onTouchCancel: PropTypes.func,
  onTouchCancelCapture: PropTypes.func,
  onTouchEnd: PropTypes.func,
  onTouchEndCapture: PropTypes.func,
  onTouchMove: PropTypes.func,
  onTouchMoveCapture: PropTypes.func,
  onTouchStart: PropTypes.func,
  onTouchStartCapture: PropTypes.func,
  pointerEvents: PropTypes.oneOf([ 'auto', 'box-none', 'box-only', 'none' ]),
  style: StyleSheetPropType(ViewStylePropTypes)
};

module.exports = ViewPropTypes;
