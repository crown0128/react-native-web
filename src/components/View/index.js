import { pickProps } from '../../modules/filterObjectProps'
import CoreComponent from '../CoreComponent'
import React, { PropTypes } from 'react'
import ViewStylePropTypes from './ViewStylePropTypes'

const viewStyleKeys = Object.keys(ViewStylePropTypes)

const styles = {
  // https://github.com/facebook/css-layout#default-values
  initial: {
    alignItems: 'stretch',
    borderWidth: 0,
    borderStyle: 'solid',
    boxSizing: 'border-box',
    display: 'flex',
    flexBasis: 'auto',
    flexDirection: 'column',
    flexShrink: 0,
    listStyle: 'none',
    margin: 0,
    padding: 0,
    position: 'relative',
    // button reset
    backgroundColor: 'transparent',
    color: 'inherit',
    font: 'inherit',
    textAlign: 'inherit'
  }
}

class View extends React.Component {
  static propTypes = {
    _className: PropTypes.string, // escape-hatch for code migrations
    accessibilityLabel: PropTypes.string,
    accessibilityLiveRegion: PropTypes.oneOf(['assertive', 'off', 'polite']),
    accessibilityRole: PropTypes.string,
    accessible: PropTypes.bool,
    children: PropTypes.any,
    component: CoreComponent.propTypes.component,
    pointerEvents: PropTypes.oneOf(['auto', 'box-none', 'box-only', 'none']),
    style: PropTypes.shape(ViewStylePropTypes),
    testID: CoreComponent.propTypes.testID
  }

  static stylePropTypes = ViewStylePropTypes

  static defaultProps = {
    _className: '',
    accessible: true,
    component: 'div',
    style: styles.initial
  }

  render() {
    const {
      _className,
      accessibilityLabel,
      accessibilityLiveRegion,
      accessibilityRole,
      accessible,
      pointerEvents,
      style,
      testID,
      ...other
    } = this.props

    const className = `View ${_className}`.trim()
    const pointerEventsStyle = pointerEvents && { pointerEvents }
    const resolvedStyle = pickProps(style, viewStyleKeys)

    return (
      <CoreComponent
        {...other}
        aria-hidden={accessible ? null : true}
        aria-label={accessibilityLabel}
        aria-live={accessibilityLiveRegion}
        className={className}
        role={accessibilityRole}
        style={{
          ...styles.initial,
          ...resolvedStyle,
          ...pointerEventsStyle
        }}
        testID={testID}
      />
    )
  }
}

export default View
