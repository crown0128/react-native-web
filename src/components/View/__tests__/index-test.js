/* eslint-env mocha */

import { assertProps, shallowRender } from '../../../modules/specHelpers'
import assert from 'assert'
import React from 'react'

import View from '../'

suite('components/View', () => {
  test('prop "accessibilityLabel"', () => {
    assertProps.accessibilityLabel(View)
  })

  test('prop "accessibilityLiveRegion"', () => {
    assertProps.accessibilityLiveRegion(View)
  })

  test('prop "accessibilityRole"', () => {
    assertProps.accessibilityRole(View)
  })

  test('prop "accessible"', () => {
    assertProps.accessible(View)
  })

  test('prop "children"', () => {
    const children = 'children'
    const result = shallowRender(<View>{children}</View>)
    assert.equal(result.props.children, children)
  })

  test('prop "component"', () => {
    assertProps.component(View)
  })

  test('prop "pointerEvents"', () => {
    const result = shallowRender(<View pointerEvents='box-only' />)
    assert.equal(result.props.style.pointerEvents, 'box-only')
  })

  test('prop "style"', () => {
    assertProps.style(View)
  })

  test('prop "testID"', () => {
    assertProps.testID(View)
  })
})
