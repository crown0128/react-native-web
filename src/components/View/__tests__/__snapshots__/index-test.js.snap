// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/View prop "children" 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
>
  <div
    className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
    data-testid="1"
  />
</div>
`;

exports[`components/View prop "pointerEvents" 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-pointerEvents-ah5dr5 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
/>
`;

exports[`components/View rendered element is a "div" by default 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
/>
`;

exports[`components/View rendered element is a "span" when inside <View accessibilityRole="button" /> 1`] = `
<button
  className="rn-alignItems-1oszu61 rn-appearance-30o5oe rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
  role="button"
  type="button"
>
  <span
    className="rn-alignItems-1oszu61 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-font-1lw9tu2 rn-listStyle-1ebb2ja rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-textAlign-1ttztb7 rn-textDecoration-bauka4"
  />
</button>
`;
